i18n-helpers/fuzz/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+use mdbook::book::{Book, BookItem, Chapter};
+use polib::catalog::Catalog;
+use polib::message::Message;
+use polib::metadata::CatalogMetadata;
+use std::path::PathBuf;
+
+/// Generate a random Catalog for fuzzing.
+pub fn create_catalog(translations: Vec<(&str, &str)>) -> Catalog {
+    let mut catalog = Catalog::new(CatalogMetadata::new());
+    for (idx, (msgid, msgstr)) in translations.iter().enumerate() {
+        let message = Message::build_singular()
+            .with_source(format!("foo.md:{idx}"))
+            .with_msgid(String::from(*msgid))
+            .with_msgstr(String::from(*msgstr))
+            .done();
+        catalog.append_or_update(message);
+    }
+    catalog
+}
+
+/// Generate a random Book for fuzzing.
+///
+/// The first u32 value in the tuple determines the type
+/// of a BookItem to insert into the book. The generated Book
+/// is not a fully valid book and contains only fields
+/// which are important for fuzzing.
+pub fn create_book(book_items: Vec<(u32, &str, &str)>) -> Book {
```

Comment:
This function can totally work, but it might be interesting for you to play with implementing `Arbitrary` for the `BookItem` type, see the [fuzzing documentation](https://rust-fuzz.github.io/book/cargo-fuzz/structure-aware-fuzzing.html).

Since we don't own the `BookItem` type, you will need to create a small wrapper type and then generate an arbitrary instance of this — and finally implement [`Into<mdbook::BookItem> for BookItem`](https://doc.rust-lang.org/stable/std/convert/trait.Into.html) where the second `BookItem` is your wrapper or shadow type.

Reply:
Thanks for suggestion, using `Into` in conjunction with `Arbitrary` macro makes it easier to read.